components/webgpu/lib.rs

```diff
@@ -950,6 +991,38 @@ impl<'a> WGPU<'a> {
                         }
                         gfx_select!(buffer_id => global.buffer_unmap(buffer_id));
                     },
+                    WebGPURequest::UpdateWebRenderData {
+                        buffer_id,
+                        buffer_size,
+                        external_id,
+                    } => {
+                        let global = &self.global;
+                        let data_pt = gfx_select!(buffer_id =>
+                            global.buffer_get_mapped_range(buffer_id, 0, None));
+                        let data = unsafe { slice::from_raw_parts(data_pt, buffer_size) }.to_vec();
+                        if let Some(present_data) =
+                            self.wgpu_image_map.lock().unwrap().get_mut(&external_id)
+                        {
+                            present_data.data = data;
```

Comment:
This may be a bit of a problem. Does WR know about your buffer stride?
If not, we should be "re-striding" the data as we read it from the buffer into `data` vector.

Reply:
Yes WR knows about the buffer stride. I have tested this and it works fine.